File: examples/python/basic_usage.py
import sys
import os
import time
import random

# Ensure we can import the SDK
sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '../../sdks/python')))

from hyperspace.client import HyperspaceClient

def main():
    print("Connecting to HyperspaceDB...")
    client = HyperspaceClient(host="localhost:50051", api_key="I_LOVE_HYPERSPACEDB")

    col_name = "py_sdk_test"
    
    # Clean up potentially stale collection
    client.delete_collection(col_name)
    
    print(f"Creating collection '{col_name}'...")
    if client.create_collection(col_name, 8, "l2"):
        print("Collection created.")
    else:
        print("Failed to create collection.")
        return

    print("Inserting vectors...")
    for i in range(10):
        vector = [0.1 * i] * 8
        meta = {"category": "test", "val": str(i)}
        success = client.insert(id=i, vector=vector, metadata=meta, collection=col_name)
        if not success:
            print(f"Failed to insert {i}")

    print("Vectors inserted. Waiting for async index...")
    time.sleep(1.0) 

    print("Searching...")
    query = [0.1] * 8
    results = client.search(vector=query, top_k=5, collection=col_name)
    
    print("Results:")
    for res in results:
        print(f"  ID: {res['id']}, Distance: {res['distance']:.4f}")

    # cleanup
    client.delete_collection(col_name)
    
    # ---------------------------------------------------------
    # Cognitive Math SDK (Spatial AI) Showcase
    # ---------------------------------------------------------
    print("\n--- Cognitive Math SDK ---")
    try:
        from hyperspace.math import local_entropy
        
        # Suppose an LLM agent generated this "thought" vector
        thought_vector = [0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8]
        
        # We retrieve context neighbors from HyperspaceDB
        neighbors = [
            [0.11, 0.19, 0.31, 0.4, 0.5, 0.6, 0.7, 0.8],
            [0.09, 0.21, 0.29, 0.4, 0.5, 0.6, 0.7, 0.8],
        ]
        
        entropy = local_entropy(candidate=thought_vector, neighbors=neighbors, c=1.0)
        print(f"Agent's Thought Entropy: {entropy:.4f}")
        if entropy > 0.8:
            print(" -> Warning: High hallucination probability!")
        else:
            print(" -> Thought is coherent with context.")
    except ImportError:
        print("hyperspace.math not found. Make sure you have the latest SDK version installed.")

    print("\nDone.")

if __name__ == "__main__":
    main()

File: examples/ts/basic_usage.ts
import { HyperspaceClient } from '../../sdks/ts/src/client';
import { localEntropy } from '../../sdks/ts/src/math';

async function main() {
    const client = new HyperspaceClient('localhost:50051', 'I_LOVE_HYPERSPACEDB');
    const colName = 'ts_sdk_example';

    try {
        console.log(`Creating collection '${colName}'...`);
        // Cleanup if exists
        await client.deleteCollection(colName).catch(() => { });

        await client.createCollection(colName, 8, 'l2');
        console.log('Collection created.');

        console.log('Inserting vectors...');
        for (let i = 0; i < 10; i++) {
            const vec = Array(8).fill(0.1 * i);
            // Note: Metadata support currently pending serialization fix
            await client.insert(i, vec, undefined, colName);
        }
        console.log('Insertion complete.');

        console.log('Waiting for index...');
        await new Promise(r => setTimeout(r, 1000));

        console.log('Searching...');
        const query = Array(8).fill(0.1); // Matches ID 1
        const results = await client.search(query, 5, colName);

        console.log('Results:');
        results.forEach(r => console.log(`  ID: ${r.id}, Distance: ${r.distance.toFixed(4)}`));

        await client.deleteCollection(colName);

        // --- Cognitive SDK Showcase ---
        console.log('\n--- Cognitive Math SDK ---');
        try {
            // Suppose an LLM agent generated this "thought" vector
            const thoughtVector = Array(8).fill(0.123);
            const contextNeighbors = [
                Array(8).fill(0.12),
                Array(8).fill(0.13),
            ];

            // Using math module from the new SDK
            const entropy = localEntropy(thoughtVector, contextNeighbors);

            console.log(`Agent's Thought Entropy: ${entropy.toFixed(4)}`);
            if (entropy > 0.8) {
                console.log(' -> Warning: High hallucination probability!');
            } else {
                console.log(' -> Thought is coherent with context in hyperbolic space.');
            }
        } catch (e) {
            console.log('Note: Ensure hyperspace math/agents modules are built and available.');
        }

        client.close();
    } catch (e) {
        console.error('Error:', e);
        client.close();
    }
}

main();
